### PR TITLE
protect DAClusterizerInZ_vect.cc from a too small value

### PR DESCRIPTION
**Pull request by VinInn**
as proposed by original author
see
https://hypernews.cern.ch/HyperNews/CMS/get/recoTracking/1659/4/2/1.html

### CODE REVIEW

**Comment by VinInn** [comment]
@cmsbuild , please test

**Comment by VinInn** [comment]
bugfix

File: RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc [hunk 1]
```diff
@@ -193,7 +193,7 @@ double DAClusterizerInZ_vect::update(double beta, track_t & gtracks,
     if (!useRho0)
       sumpi += gtracks._pi[itrack];
     
-    if (gtracks._Z_sum[itrack] > 0) {
+    if (gtracks._Z_sum[itrack] > 1.e-100) {
       kernel_calc_normalization(itrack, gtracks, gvertices);
     }
```
Comment (cvuosalo) [inline RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc]: Please let's stop the propagation of the bad value by adding:
```
else if (isNotFinite((gtracks._Z_sum[itrack]))
  gtracks._Z_sum[itrack] = 0.0;
```

**Comment by cvuosalo** [comment]
Tests of workflows 136.731_RunSinglePh2016B and 25202.0_TTbar_13 with 1000 and 100 events, respectively, against baseline CMSSW_9_0_X_2016-11-25-1100 show a large number of differences in values related to hits, tracks, jets, and MET, but these differences are very tiny and insignificant.

**Comment by VinInn** [comment]
@cmsbuild , please test

**Comment by VinInn** [comment]
@cmsbuild , please test

**Comment by cvuosalo** [comment]
+1

For #16761 99eec6a0241abd8b1c7fdbbb1d1bfda3570bc9b7

Fixing a bug by not processing NAN values in the PrimaryVertexProducer.

#16762 and #16764 are the 81X and 80X versions of this PR. The former has already been approved.

The code changes are satisfactory. Jenkins tests against baseline CMSSW_9_0_X_2016-11-25-2300 show a large number of differences in values related to hits, tracks, jets, and MET, but these differences are very tiny and insignificant. More extensive testing described above (https://github.com/cms-sw/cmssw/pull/16761#issuecomment-263039751) also shows the presence of these tiny differences, but none are significant.

**Pull request merged**